cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/queue/AbstractQueueSpecificationGenerator.java

```diff
@@ -72,7 +72,7 @@
       }
 
       if (connection.getSourceType() == FlowletConnection.Type.STREAM) {
-        builder.add(createSpec(QueueName.fromStream(outputName),
+        builder.add(createSpec(QueueName.fromStream(app.getNamespaceId(), outputName),
```

Comment:
Shouldn't this be ``Id.Stream`` too instead of ``QueueName``?

If we're planning on using ``Id.Stream`` always, it may make sense to remove the ``fromStream`` and other *stream* methods from ``QueueName`` so you can spot these parts?